runtime/interface.go

```diff
@@ -69,10 +52,16 @@ type Interface interface {
 	SetValue(owner, key, value []byte) (err error)
 	// CreateAccount creates a new account.
 	CreateAccount(payer Address) (address Address, err error)
+	// AddEncodedAccountKey appends an encoded key to an account.
+	AddEncodedAccountKey(address Address, publicKey []byte) error
+	// RemoveEncodedAccountKey removes a key from an account by index, add returns the encoded key.
+	RemoveEncodedAccountKey(address Address, index int) (publicKey []byte, err error)
 	// AddAccountKey appends a key to an account.
-	AddAccountKey(address Address, publicKey []byte) error
+	AddAccountKey(address Address, publicKey *PublicKey, hashAlgo HashAlgorithm, weight int) (*AccountKey, error)
+	// GetAccountKey retrieves a key from an account by index.
+	GetAccountKey(address Address, index int) (*AccountKey, error)
 	// RemoveAccountKey removes a key from an account by index.
-	RemoveAccountKey(address Address, index int) (publicKey []byte, err error)
+	RemoveAccountKey(address Address, index int) (*AccountKey, error)
```

Comment:
Given that the key is not removed, but revoked, maybe rename this to `RevokeAccountKey` (The Cadence function is also named `revoke`)